### PR TITLE
Fix for regexp queries through pihole -q

### PR DESCRIPTION
**Pull request by mmotti**
Taking inspiration from @WaLLy3K (https://github.com/pi-hole/pi-hole/pull/2346)

I have taken his changes and amended slightly to allow for this to work properly.

Creating a new PR as the existing one is behind on the current commits and I am unsure on how to merge my changes.

**Output**
```
pi@raspberrypi:~ $ pihole -q track.ads.test.com
 2 matches found in regex table
   ^track(ers?|ing)?[0-9]*[-.]
   ^(.+[-_.])??m?ad[sxv]?[0-9]*[-_.]

pi@raspberrypi:~ $ pihole -q ads.test.com
 1 match found in regex table
   ^(.+[-_.])??m?ad[sxv]?[0-9]*[-_.]
```

**By submitting this pull request, I confirm the following:** 
*please fill any appropriate checkboxes, e.g: [X]*

- [ x ] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [ x ] I have made only one major change in my proposed changes.
- [ x ] I have commented my proposed changes within the code.
- [ x ] I have tested my proposed changes, and have included unit tests where possible.
- [ x ] I am willing to help maintain this change if there are issues with it later.
- [ x ] I give this submission freely and claim no ownership.
- [ x ] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [ x ] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

Please make sure you [Sign Off](https://github.com/pi-hole/pi-hole/wiki/How-to-signoff-your-commits.) all commits. Pi-hole enforces the [DCO](https://github.com/pi-hole/pi-hole/wiki/Contributing-to-the-project).

---
**What does this PR aim to accomplish?:**
Allow `pihole -q` to return matches in regex.list

**How does this PR accomplish the above?:**
Instead of a two-way grep, we instead use awk to iterate through each regexp in regex.list and print any lines that match our domain query.

**What documentation changes (if any) are needed to support this PR?:**
N/A

---
* You must follow the template instructions. Failure to do so will result in your pull request being closed.
* Please respect that Pi-hole is developed by volunteers, who can only reply in their spare time.

### CODE REVIEW

File: advanced/Scripts/query.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # Globals
 piholeDir="/etc/pihole"
 gravityDBfile="${piholeDir}/gravity.db"
-wildcardlist="/etc/dnsmasq.d/03-pihole-wildcard.conf"
+regexlist="/etc/pihole/regex.list"
```
Comment (Mcat12) [inline advanced/Scripts/query.sh]: This is no longer needed

**Comment by mmotti** [comment]
I think that should mostly cover it

File: advanced/Scripts/query.sh [hunk 2]
```diff
@@ -145,24 +132,32 @@ scanDatabaseTable() {
...
+str_regexlist=$(IFS=$'\n'; echo "${regexlist[*]}")
+# If there are regexps in the DB
+if [[ -n "${str_regexlist}" ]]; then
+    # Return any regexps that match the domainQuery
+    mapfile -t results <<< "$(scanList "${domainQuery}" "${str_regexlist}" "regex")"
+
+	# If there are matches to the domain query
+	if [[ -n "${results[*]}" ]]; then
+		# Form output strings
+		str="${matchType^} found in ${COL_BOLD}regex list${COL_NC}"
```
Comment (Mcat12) [inline advanced/Scripts/query.sh]: Minor nit: capitalize `regex` to fit with how `Whitelist` and `Blacklist` are capitalized.

Comment (mmotti) [reply advanced/Scripts/query.sh]: Update this and the "bp" output

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/faulty-regex-consumes-all-memory-and-causes-issues-with-pihole-q/28710/41

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/expand-query-lists-to-regex/11915/4

**Pull request merged**